### ISSUE TITLE
Exception when running from source with `npm start`

### ISSUE THREAD

**Issue by tofi86** [edrlab/readium-desktop#135]
I followed the README to the point but get an exception on startup when running `npm start`:

![bildschirmfoto 2018-04-02 um 13 47 38](https://user-images.githubusercontent.com/365118/38195180-782daa7c-367c-11e8-8892-5c8feab49772.png)

I cannot close the dialog with "OK" button and need to force-close the process with Ctrl+C in Terminal.

Running `npm rebuild` or `npm install` again doesn't solve the issue.

This is the last lines from console log:

```
Child renderer webview preload:
    Hash: 68d747b443a481ba2218
    Time: 5773ms
         Asset     Size  Chunks             Chunk Names
    preload.js  61.3 kB       0  [emitted]  main
       13 modules
/Users/tofi/Entwicklung/git-workspace/readium-desktop/node_modules/r2-navigator-js/dist/ReadiumCSS
App threw an error during load
Error: The module '/Users/tofi/Entwicklung/git-workspace/readium-desktop/node_modules/leveldown/build/Release/leveldown.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 59. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at bindings (/Users/tofi/Entwicklung/git-workspace/readium-desktop/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/Users/tofi/Entwicklung/git-workspace/readium-desktop/node_modules/leveldown/leveldown.js:4:46)
    at Object.<anonymous> (/Users/tofi/Entwicklung/git-workspace/readium-desktop/node_modules/leveldown/leveldown.js:109:3)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
```

Running `npm run start:dev` works fine though...

My env:

* Mac OS X: 10.13.4
* NodeJS: v9.10.1
* npm: v5.6.0

**Comment by tofi86** [comment]
Hm, nevermind. `npm update` fixed the issue. Have no clue why...

**Comment by danielweck** [comment]
NPM's 'post install' is currently prefixed with two underscores, so that the script does not execute every time. Therefore, the required step of building native modules (leveldown) must be invoked manually, for example:
'npm run install && npm run __postinstall && npm run start'
Note that you can also use 'npm run start:dev' to enable debugging, in which case a different database implementation is used under the hood.

package.json '__postinstall' script:
https://github.com/edrlab/readium-desktop/blob/develop/package.json

**Comment by tofi86** [comment]
Hi Daniel,

I get the following when running your commandline `npm run install && npm run __postinstall && npm run start`:

```
npm ERR! missing script: install
```

But running `npm run __postinstall && npm run start` alone worked fine.

**Comment by danielweck** [comment]
Sorry about the typo, I meant 'npm install' not 'npm run install' :)

**Comment by tofi86** [comment]
Ah well, that sounds reasonable ;-) 👍 
Sorry, I'm a total NodeJS n00b  :dizzy_face:

**Comment by JayPanoz** [comment]
Hi @tofi86, if that can help – that’s an “issue” I came across with readium-css and tried to document because, well, npm usage is quite unusual for anything CSS and can put a barrier to entry :-/

- [npm install](https://docs.npmjs.com/cli/install), installs a package, and any packages that it depends on;
- [npm run <stage>](https://docs.npmjs.com/cli/run-script), actually an alias of `npm run-script` so makes use of an arbitrary script you’ll find in the `scripts` field of `package.json`;
- [npm test](https://docs.npmjs.com/cli/test), actually an alias of `npm run-script test` so makes use of an arbitrary script you’ll find in the `scripts` field of `package.json`. If the package doesn’t have one, the command will fail;
- [how nom handles the `scripts` field](https://docs.npmjs.com/misc/scripts).

TL;DR: arbitrary scripts can be executed by running `npm run-script <stage>`.

**Comment by tofi86** [comment]
Thanks, Jiminy! 👍
